File: leetcode/01-two-sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const memo = {};

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];

    if (diff in memo) {
      return [i, memo[diff]];
    }

    memo[nums[i]] = i;
  }
};

File: leetcode/01-two-sum/twoSum.go
package main

func twoSum(nums []int, target int) []int {
    memo := make(map[int]int)

    for index, num := range nums {
        if val, exists := memo[target-num]; exists {
            return []int {val, index}
        }

        memo[num] = index
    }

    return nil
}